samples/deploy/crd/v1beta1/ApisixRoute.yaml

```
@@ -35,3 +48,146 @@ spec:
     kind: ApisixRoute
     shortNames:
       - ar
+  preserveUnknownFields: false
+  validation:
+    openAPIV3Schema:
+      type: object
+      properties:
+        spec:
+          type: object
+          properties:
+            http:
+              type: array
+              minItems: 1
+              items:
+                type: object
+                oneOf:
+                  - required: ["name", "match", "backend"]
+                  - required: ["name", "match", "backends"]
+                properties:
+                  name:
+                    type: string
+                    minLength: 1
+                  priority:
+                    type: integer
+                  match:
+                    type: object
+                    required:
+                    - paths
+                    properties:
+                      paths:
+                        type: array
+                        minItems: 1
+                        items:
+                          type: string
+                          pattern: "^/" # start with "/"
+                      hosts:
+                        type: array
+                        minItems: 1
+                        items:
+                          type: string
+                          pattern: "^\\*?[0-9a-zA-Z-._]+$"
+                      methods:
+                        type: array
+                        minItems: 1
+                        items:
+                          type: string
+                          enum: ["GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS", "CONNECT", "TRACE"]
+                      remoteAddrs:
+                        type: array
+                        minItems: 1
+                        items:
+                          type: string
+                      exprs:
+                        type: array
+                        minItems: 1
+                        items:
+                          type: object
+                          properties:
+                            subject:
+                              type: object
+                              properties:
+                                scope:
+                                  type: string
+                                  enum: ["Header", "Query", "Cookie", "Path"]
```

Should enums be sorted consistently the same way in each file ?

```suggestion
                                  enum: ["Cookie", "Header", "Path", "Query"]
```

---

Good point.